pkg/engine/patches.go

```diff
@@ -14,7 +16,7 @@ type PatchBytes []byte
 // ProcessPatches Returns array from separate patches that can be applied to the document
 // Returns error ONLY in case when creation of resource should be denied.
 func ProcessPatches(rule kubepolicy.Rule, resource []byte) (allPatches []PatchBytes, errs []error) {
-	if len(resource) == 0 {
+	if len(resource) == 0 || rule.Mutation == nil {
 		errs = append(errs, errors.New("Source document for patching is empty"))
```

Comment:
We need to split these two scenarios as error message is not clear here. 
See https://github.com/nirmata/kyverno/blob/1f8b370f5a0038cb8c0e4a7505546dcf889876e0/pkg/engine/patches.go#L19

Reply:
updated